src/components/Todos.tsx

```diff
@@ -0,0 +1,147 @@
+    event.preventDefault();
+    if (newTitle === '') {
+      onDelete(todoId);
+    } else {
+      changeTodoTitle(todoId, newTitle);
+    }
+
+    setEditingTodo(null);
+  }
+
+  function handleEditBlur(todoId: number) {
+    if (newTitle === '') {
+      onDelete(todoId);
+    } else {
+      changeTodoTitle(todoId, newTitle);
+    }
+
+    setEditingTodo(null);
+    setDoubleClicked(false);
+  }
+
+  function handleEditKey(event: React.KeyboardEvent, todoId: number) {
+    if (event.key === 'Escape') {
+      setNewTitle('');
+      setEditingTodo(null);
+    }
+
+    if (event.key === 'Enter') {
+      changeTodoTitle(todoId, newTitle);
+    }
+  }
+
+  return (
+    <section className="todoapp__main">
+      {todos.map((todo) => (
+        <div
+          className={classNames('todo', {
+            completed: todo.completed,
+          })}
+          key={todo.id}
+        >
+          {editingTodo === todo.id ? (
+            <>
+              <label
+                className="todo__status-label"
+              >
+                <input
+                  type="checkbox"
+                  className="todo__status"
+                  checked={todo.completed}
+                  onChange={() => todoStatus(todo.id)}
+                />
+              </label>
+              <form onSubmit={(event) => handleEditSubmit(event, todo.id)}>
+                <input
+                  type="text"
+                  className="todo__title-field"
+                  value={newTitle}
+                  onChange={(e) => setNewTitle(e.target.value)}
+                  onBlur={(e) => {
+                    if (e.relatedTarget === null) {
+                      handleEditBlur(todo.id);
+                    }
+                  }}
+                  // eslint-disable-next-line jsx-a11y/no-autofocus
+                  autoFocus={doubleClicked}
+                  onKeyUp={(e) => handleEditKey(e, todo.id)}
+                />
+              </form>
+            </>
+          ) : (
+            <>
+              <label
+                className="todo__status-label"
+              >
+                <input
+                  type="checkbox"
+                  className="todo__status"
+                  checked={todo.completed}
+                  onChange={() => todoStatus(todo.id)}
+                />
+              </label>
+
+              <span
+                className="todo__title"
+                onDoubleClick={() => setNewTodoTitle(todo.id, todo.title)}
+              >
+                {todo.title}
+              </span>
+
+              <button
+                type="button"
+                className="todo__remove"
+                onClick={() => onDelete(todo.id)}
+              >
+                ×
+              </button>
+            </>
+          )}
+
+          <div className="modal overlay">
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
```

Comment:
you should also show the loader for any change - editing the todo title, checking/unchecking, etc (including batch actions - clear completed).
See [working example](https://mate-academy.github.io/react_todo-app-with-api/)

Generally, I guess it might be optionally, but I'd strongly recommend do it - it will look better, at least before adding to the portfolio